### ISSUE TITLE
Can't Update My Sub On The Workshop

### ISSUE THREAD

**Issue by Viperspider** [Regalis11/Barotrauma#10636]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I can't update my workshop sub. I have a local copy which I have updated and want to upload, but any time I hit the "update" button in the mods menu it just uploads the old version (IE: The version already on the workshop). I have tried everything from manually overwriting the file in /localmods/ to copy and pasting the damn new version over the old (not viable).

I believe this issue occurred after I unistalled the game and reinstalled it at a later date. I believe I would have downloaded the Megalodon from the workshop and updated it that way, but I may as well have copy and pasted a local backup of the Megalodon into the game directory before editing that. Probably the former though. 

How can I share my updated sub?

My sub from /localmods/: [Megalodon.zip](https://github.com/Regalis11/Barotrauma/files/10135747/Megalodon.zip)

My sub on the workshop: https://steamcommunity.com/sharedfiles/filedetails/?id=2769582416

I have attached what I believe is the most recent version of my sub. Should be pretty easy to tell, as the workshop version will have no large weapon hardpoints and the new one should be updated to include all the features of the latest update.

### Reproduction steps

1. Create a sub.
2. Upload it to the workshop
3. Uninstall the game.
4. Reinstall after an update.
5. Download your sub again.
6. Edit it.
7. Try to update the sub on the workshop.
8. Cry, as your effort is wasted.

### Bug prevalence

Happens every time I play

### Version

0.19.14.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Unless there's upload logs, I don't think there's anything I can provide here.
```

**Comment by Regalis11** [comment]
Thank you for the report!

> but I may as well have copy and pasted a local backup of the Megalodon into the game directory before editing that

Looking at the file you provided, this seems like the most probable cause: the filelist.xml file doesn't include the Workshop ID of the mod, so the game has no way of determining which Workshop mod is related to, and essentially just treats it as any sub you've only saved locally.

To be able to publish updates to the sub, you should instead download the submarine from the workshop. What I recommend doing now:

1. Make a manual backup of the latest version of the submarine file (and save it somewhere safe, outside the game folder to be sure it doesn't get modified when you download the mod)
2. Delete the whole submarine from your local mods (not just the .sub file, but the entire "Megalodon" folder)
3. Download the mod from the workshop and enable it
4. Find the local copy of the mod from LocalMods, and replace the submarine file (just the submarine file, Megalodon.sub) with the manual backup you made previously.

**Comment by Viperspider** [comment]
Thanks Jonas, where can I find the Workshop mod in the game files?

The problem I had before when trying this was that I could only replace the local versions of the sub, and when I went to upload the update my local changes were not being uploaded to the workshop when I tried to publish.

Is there any way I can directly replace the workshop download without having to make a local copy of the workshop sub?

Best Regards,
Shane Egan.
